xds/internal/xdsclient/authority_test.go

```diff
@@ -267,12 +273,127 @@ func (s) TestWatchResourceTimerCanRestartOnIgnoredADSRecvError(t *testing.T) {
 
 }
 
+// This tests the scenario when IgnoreResourceDeletion server_features variable is set on
+// the bootstrap server config, the resource deletion is ignored by the client when a update
+// is missing that resource. This is a conformance gRFC A53[https://github.com/grpc/proposal/blob/master/A53-xds-ignore-resource-deletion.md].
+func (s) TestResourceDeletionIsIgnored(t *testing.T) {
```

Comment:
I would really prefer to see an e2e test for this instead, since the behavior can be tested quite effectively using an e2e test without relying on internal state.

I'd like to see three tests (all three should have two versions, one with `ignore_resource_deletion` feature set to `true` and other with it being set to `false`)
- Listener resource is deleted on the client
  - In this case, let's make sure that the route configuration pointed to two cluster resources (easier to do based on rpc path, but via rpc headers are also fine).
  - And the test would verify that we can make RPCs to both clusters
  - then the listener resource would be removed
  - then, we verify that RPCs can still be make to both the clusters. In the case where `ignore_resource_deletion` is `false`, RPCs to both clusters should fail
- Cluster resource is deleted on the client
  - Similar to the above case where we have two clusters
  - After removing one of the cluster resources, verify that RPCs to both clusters work fine. In the case where `ignore_resource_deletion` is `false`, RPC to one cluster should work, while RPC to the deleted one should fail
- Listener resource is deleted on the server
  - Setup server, and ensure everything is working e2e
  - Remove the listener resource 
  - When `ignore_resource_deletion` is set to `true`, server should continue to serve requests.
  - When `ignore_resource_deletion` is set to `false`, server should enter `non_serving` mode

Reply:
uff. This took a while.. Its done.